### ISSUE TITLE
Add ability to delete specific seasons without removing request/Sonarr data

### ISSUE THREAD

**Issue by Ultragian** [Fallenbagel/jellyseerr#585]
### Description

I like that users can request media, I like that I can delete media too. The only issue that I can't delete specific seasons without deleting whole shows, unless I go directly into Sonarr.
![image](https://github.com/Fallenbagel/jellyseerr/assets/7107755/2051c9b6-896f-49e3-8350-5202c3d33c71)

### Desired Behavior

Delete specific seasons directly from the Request menu without removing request data from Jellyseerr and at the same time being able to choose whether to remove the "monitored" tag from either Sonarr or Radarr(that way they don't just redownload it straight away).
Either add a new improved delete menu or make the Request menu both a request and remove menu (the latter seeming to be easier).

### Additional Context

My family can request data, they can't delete data though. They message me to delete seasons they are done with, but even I have to manually do that in Sonarr (episode by episode).
Everyone should be able to control it all through Jellyseerr.

### Code of Conduct

- [X] I agree to follow Jellyseerr's Code of Conduct

**Comment by Fallenbagel** [comment]
The new #522 media availability sync job should clear seasons if lets say you delete season 5 from both sonarr and jellyfin/emby that season will be removed by that job.

**Comment by gauthier-th** [comment]
@Ultragian can this be closed?

**Comment by Ultragian** [comment]
Sorry for the late reply, but I don't agree on this being closed as the feature was never implemented.
I want to be able to also delete specific seasons directly in Jellyseerr. If I can add specific seasons, why can't I remove specific seasons too?
The sync feature mentioned above doesn't replace this, as that just depends on a third-party app to do the job.
I see it as being able to control almost everything through Jellyseerr. That way authorized users who don't understand much wouldn't have to ask me directly to remove a show.

**Comment by rVlad93** [comment]
Just starting using it but I agree, this would be a welcome addition.

**Comment by botagas** [comment]
Chiming in as well - Jellyseerr is the main page accessed by most users on a media server. If I request some media on it, I also expect to be able to delete that media. If I liked the series, I may want to remove the season I watched, but keep the request for future seasons. At the moment, I don't see this ability yet.

**Comment by fallenbagel** [comment]
> Delete their own requested content (whole or partial deletion, like a season or two) if permission is granted

I don't think we will extend content deletion permission further from `manage request` users.

> Request deletion of content (whole or partial deletion, like a season or two)

#308
